### PR TITLE
Migrate UserTest to the new base testsuite

### PR DESCRIPTION
**Pull request by miquelsi**
Part of #31750

### CODE REVIEW

**Review by vaceksimon — COMMENTED** [review]
Overall, it doesn't seem to me, this is a good test class to showcase the functionality of the new framework. If we want to show the benefits of this PoC, this class sets a limit to what we can use, as it relies on creating the users by itself.

I would reconsider how we approach this test and choose another one for #31312

**Review by lhanusov — COMMENTED** [review]
I think this may be a good candidate for Mock and Provider unit testing. WDYT?@stianst

**Review by lhanusov — APPROVED** [review]
LGTM

File: test-poc/showcase/src/test/java/org/keycloak/test/showcase/UserTest.java [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+    ManagedRealm realm;
+
+    @InjectOAuthClient
+    OAuthClient oAuthClient;
+
+    @InjectWebDriver
+    WebDriver webDriver;
+
+    @InjectPage
+    LoginPage loginPage;
```
Comment (stianst) [inline test-poc/showcase/src/test/java/org/keycloak/test/showcase/UserTest.java:46]: Would remove these as they are not currently used, and honestly using these should not be done in a test that tests admin endpoints. The commented out test later should really be separated out into a separate place.

File: test-poc/showcase/src/test/java/org/keycloak/test/showcase/UserTest.java [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+
+    @AfterEach
+    public void afterEach() {
+        realm.admin().identityProviders().findAll()
+                .forEach(ip -> realm.admin().identityProviders().get(ip.getAlias()).remove());
+        realm.admin().groups().groups()
+                .forEach(g -> realm.admin().groups().group(g.getId()).remove());
+        realm.admin().users().list()
+                .forEach(u -> realm.admin().users().delete(u.getId()));
```
Comment (stianst) [inline test-poc/showcase/src/test/java/org/keycloak/test/showcase/UserTest.java:55]: This is fairly stupid thing to do; guessing it's just a copy/paste of sorts from the existing test?

File: test-poc/showcase/src/test/java/org/keycloak/test/showcase/UserTest.java [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+
+        getCleanup().addUserId(createdId);*/
+
+        return createdId;
+    }
+
+    @Test
+    public void verifyCreateUser() {
+        createUser();
+    }
```
[the remark this answers is not in the document]
Comment (stianst) [reply test-poc/showcase/src/test/java/org/keycloak/test/showcase/UserTest.java:99]: Nah, this test should really not be using managed users, it should be using the admin client directly.

**Comment by stianst** [comment]
> Overall, it doesn't seem to me, this is a good test class to showcase the functionality of the new framework. If we want to show the benefits of this PoC, this class sets a limit to what we can use, as it relies on creating the users by itself.
> 
> I would reconsider how we approach this test and choose another one for #31312

On the contrary - this test is perfect as it can simply just be moved with not much modification :)

**Comment by stianst** [comment]
> I think this may be a good candidate for Mock and Provider unit testing. WDYT?@stianst

Not really - this is about testing the actual admin-endpoints. However, there are probably some tests within here that doesn't belong here.

**Review by vaceksimon — APPROVED** [review]

**Review by stianst — APPROVED** [review]

**Pull request merged**